--- origsrc/kpeople-5.12.0/src/backends/CMakeLists.txt	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/backends/CMakeLists.txt	2015-07-21 18:59:24.490007300 -0500
@@ -13,7 +13,9 @@ target_link_libraries (KF5PeopleBackend
                         Qt5::Widgets #needed for QAction, otherwise QtGui
 )
 
-target_include_directories (KF5PeopleBackend INTERFACE "$<INSTALL_INTERFACE:${KF5_INCLUDE_INSTALL_DIR}/KPeople>")
+target_include_directories (KF5PeopleBackend PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/kpeoplebackend>"
+                                             INTERFACE "$<INSTALL_INTERFACE:${KF5_INCLUDE_INSTALL_DIR}/KPeople>"
+)
 set_target_properties (KF5PeopleBackend PROPERTIES VERSION ${KPEOPLE_VERSION_STRING} SOVERSION ${KPEOPLE_SOVERSION} EXPORT_NAME PeopleBackend)
 
 install (TARGETS KF5PeopleBackend
@@ -36,6 +38,9 @@ install (FILES ${KPeopleBackend_CamelCas
          COMPONENT Devel
 )
 install(FILES
+  ${CMAKE_CURRENT_BINARY_DIR}/kpeoplebackend/kpeoplebackend_export.h
   ${KPeopleBackend_HEADERS}
   DESTINATION ${KF5_INCLUDE_INSTALL_DIR}/KPeople/kpeoplebackend COMPONENT Devel
 )
+
+generate_export_header(KF5PeopleBackend EXPORT_FILE_NAME ${CMAKE_CURRENT_BINARY_DIR}/kpeoplebackend/kpeoplebackend_export.h BASE_NAME KPeopleBackend)
--- origsrc/kpeople-5.12.0/src/backends/abstractcontact.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/backends/abstractcontact.h	2015-07-21 18:35:22.483895800 -0500
@@ -21,7 +21,7 @@
 
 #include <QSharedData>
 #include <QVariant>
-#include <kpeople/kpeople_export.h>
+#include <kpeoplebackend/kpeoplebackend_export.h>
 
 namespace KPeople
 {
@@ -35,7 +35,7 @@ namespace KPeople
  * @internal
  */
 
-class KPEOPLE_EXPORT AbstractContact : public QSharedData
+class KPEOPLEBACKEND_EXPORT AbstractContact : public QSharedData
 {
 public:
     typedef QExplicitlySharedDataPointer<AbstractContact> Ptr;
--- origsrc/kpeople-5.12.0/src/backends/abstractpersonaction.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/backends/abstractpersonaction.h	2015-07-21 18:35:22.488396300 -0500
@@ -22,14 +22,14 @@
 #include <QObject>
 #include <QAction>
 
-#include <kpeople/kpeople_export.h>
+#include <kpeoplebackend/kpeoplebackend_export.h>
 #include <KPeople/PersonData>
 
 namespace KPeople
 {
 class PersonData;
 
-class KPEOPLE_EXPORT AbstractPersonAction : public QObject
+class KPEOPLEBACKEND_EXPORT AbstractPersonAction : public QObject
 {
     Q_OBJECT
 public:
--- origsrc/kpeople-5.12.0/src/backends/allcontactsmonitor.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/backends/allcontactsmonitor.h	2015-07-21 18:35:22.492896900 -0500
@@ -23,7 +23,7 @@
 #include <QObject>
 #include <QSharedPointer>
 
-#include <kpeople/kpeople_export.h>
+#include <kpeoplebackend/kpeoplebackend_export.h>
 #include <kpeoplebackend/abstractcontact.h>
 
 namespace KPeople
@@ -38,7 +38,7 @@ class AllContactsMonitorPrivate;
  * Subclasses are expected to be asynchronous
  *
  */
-class KPEOPLE_EXPORT AllContactsMonitor : public QObject
+class KPEOPLEBACKEND_EXPORT AllContactsMonitor : public QObject
 {
     Q_OBJECT
 public:
--- origsrc/kpeople-5.12.0/src/backends/basepersonsdatasource.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/backends/basepersonsdatasource.h	2015-07-21 18:35:22.497897500 -0500
@@ -23,7 +23,7 @@
 #include <QObject>
 #include <QVariant>
 
-#include <kpeople/kpeople_export.h>
+#include <kpeoplebackend/kpeoplebackend_export.h>
 
 #include "allcontactsmonitor.h"
 #include "contactmonitor.h"
@@ -34,7 +34,7 @@ namespace KPeople
 class BasePersonsDataSourcePrivate;
 
 //This is a QObject for KPluginFactory
-class KPEOPLE_EXPORT BasePersonsDataSource : public QObject
+class KPEOPLEBACKEND_EXPORT BasePersonsDataSource : public QObject
 {
     Q_OBJECT
 public:
--- origsrc/kpeople-5.12.0/src/backends/contactmonitor.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/backends/contactmonitor.h	2015-07-21 18:35:22.500897900 -0500
@@ -22,7 +22,7 @@
 
 #include <QObject>
 
-#include <kpeople/kpeople_export.h>
+#include <kpeoplebackend/kpeoplebackend_export.h>
 
 #include "abstractcontact.h"
 
@@ -39,7 +39,7 @@ class ContactMonitorPrivate;
  * Subclasses are expected to be asynchronous in loading data.
  *
  */
-class KPEOPLE_EXPORT ContactMonitor: public QObject
+class KPEOPLEBACKEND_EXPORT ContactMonitor: public QObject
 {
     Q_OBJECT
 public:
--- origsrc/kpeople-5.12.0/src/backends/defaultcontactmonitor_p.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/backends/defaultcontactmonitor_p.h	2015-07-21 18:35:22.503398200 -0500
@@ -32,7 +32,7 @@ using namespace KPeople;
  * ideally plugins (especially slow ones) will implement their own contact monitor.
 */
 
-class KPEOPLE_EXPORT DefaultContactMonitor : public ContactMonitor
+class KPEOPLEBACKEND_EXPORT DefaultContactMonitor : public ContactMonitor
 {
     Q_OBJECT
 public:
--- origsrc/kpeople-5.12.0/src/declarative/personactionsmodel_p.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/declarative/personactionsmodel_p.h	2015-07-21 19:04:45.057714200 -0500
@@ -21,15 +21,13 @@
 
 #include <QAbstractListModel>
 
-#include <kpeople/kpeople_export.h>
-
 class QAction;
 
 namespace KPeople
 {
 class PersonActionsPrivate;
 
-class KPEOPLE_EXPORT PersonActionsModel : public QAbstractListModel
+class PersonActionsModel : public QAbstractListModel
 {
     Q_OBJECT
     Q_PROPERTY(int count READ rowCount NOTIFY personChanged)
--- origsrc/kpeople-5.12.0/src/widgets/CMakeLists.txt	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/widgets/CMakeLists.txt	2015-07-21 18:50:32.372937000 -0500
@@ -65,10 +65,13 @@ install (FILES ${KPeopleWidgets_CamelCas
          COMPONENT Devel
 )
 install(FILES
+  ${CMAKE_CURRENT_BINARY_DIR}/kpeople/widgets/kpeoplewidgets_export.h
   ${KPeopleWidgets_HEADERS}
   DESTINATION ${KF5_INCLUDE_INSTALL_DIR}/KPeople/kpeople/widgets COMPONENT Devel
 )
 
+generate_export_header(KF5PeopleWidgets EXPORT_FILE_NAME ${CMAKE_CURRENT_BINARY_DIR}/kpeople/widgets/kpeoplewidgets_export.h BASE_NAME KPeopleWidgets)
+
 install (FILES persondetailsplugin.desktop
          DESTINATION
             ${SERVICETYPES_INSTALL_DIR}
--- origsrc/kpeople-5.12.0/src/widgets/abstractfieldwidgetfactory.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/widgets/abstractfieldwidgetfactory.h	2015-07-21 18:42:45.731681100 -0500
@@ -22,14 +22,14 @@
 #include <QWidget>
 #include <QIcon>
 
-#include <kpeople/kpeople_export.h>
+#include <kpeople/widgets/kpeoplewidgets_export.h>
 
 namespace KPeople
 {
 class AbstractFieldWidgetFactoryPrivate;
 class PersonData;
 
-class KPEOPLE_EXPORT AbstractFieldWidgetFactory : public QObject
+class KPEOPLEWIDGETS_EXPORT AbstractFieldWidgetFactory : public QObject
 {
     Q_OBJECT
 public:
--- origsrc/kpeople-5.12.0/src/widgets/actions.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/widgets/actions.h	2015-07-21 18:42:45.734681500 -0500
@@ -20,7 +20,7 @@
 #define KPEOPLEWIDGETS_GLOBAL_H
 
 #include <QList>
-#include <kpeople/kpeople_export.h>
+#include <kpeople/widgets/kpeoplewidgets_export.h>
 
 class QString;
 class QObject;
@@ -48,7 +48,7 @@ enum ActionType {
  * Returns a list of actions relevant to the specified @p contactUri where
  * each QAction will have @p parent passed as its parent QObject
  */
-KPEOPLE_EXPORT QList<QAction *> actionsForPerson(const QString &contactUri, QObject *parent);
+KPEOPLEWIDGETS_EXPORT QList<QAction *> actionsForPerson(const QString &contactUri, QObject *parent);
 }
 
 #endif // KPEOPLEWIDGETS_GLOBAL_H
--- origsrc/kpeople-5.12.0/src/widgets/mergedelegate.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/widgets/mergedelegate.h	2015-07-21 18:42:45.737181800 -0500
@@ -20,7 +20,7 @@
 #ifndef MERGEDELEGATE_H
 #define MERGEDELEGATE_H
 
-#include <kpeople/kpeople_export.h>
+#include <kpeople/widgets/kpeoplewidgets_export.h>
 
 #include <QItemSelection>
 #include <QItemDelegate>
@@ -29,7 +29,7 @@
 class QAbstractItemView;
 class QItemSelection;
 
-class KPEOPLE_EXPORT MergeDelegate: public KExtendableItemDelegate
+class KPEOPLEWIDGETS_EXPORT MergeDelegate: public KExtendableItemDelegate
 {
     Q_OBJECT
 
--- origsrc/kpeople-5.12.0/src/widgets/mergedialog.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/widgets/mergedialog.h	2015-07-21 18:42:45.739682100 -0500
@@ -20,7 +20,7 @@
 #ifndef MERGEDIALOG_H
 #define MERGEDIALOG_H
 
-#include <kpeople/kpeople_export.h>
+#include <kpeople/widgets/kpeoplewidgets_export.h>
 
 #include <QDialog>
 #include <QModelIndex>
@@ -41,7 +41,7 @@ class Match;
  * By properly mergeing contacts, the user will end up with having more
  * information on each person.
  */
-class KPEOPLE_EXPORT MergeDialog : public QDialog
+class KPEOPLEWIDGETS_EXPORT MergeDialog : public QDialog
 {
     Q_OBJECT
 
--- origsrc/kpeople-5.12.0/src/widgets/persondetailsdialog.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/widgets/persondetailsdialog.h	2015-07-21 18:42:45.742682500 -0500
@@ -22,14 +22,14 @@
 
 #include <QDialog>
 
-#include <kpeople/kpeople_export.h>
+#include <kpeople/widgets/kpeoplewidgets_export.h>
 #include "../persondata.h"
 
 namespace KPeople
 {
 class PersonDetailsDialogPrivate;
 
-class KPEOPLE_EXPORT PersonDetailsDialog : public QDialog
+class KPEOPLEWIDGETS_EXPORT PersonDetailsDialog : public QDialog
 {
     Q_OBJECT
 
--- origsrc/kpeople-5.12.0/src/widgets/persondetailsview.h	2015-07-05 04:11:07.000000000 -0500
+++ src/kpeople-5.12.0/src/widgets/persondetailsview.h	2015-07-21 18:42:45.745182800 -0500
@@ -22,7 +22,7 @@
 #include <QWidget>
 #include <QVariantList>
 
-#include <kpeople/kpeople_export.h>
+#include <kpeople/widgets/kpeoplewidgets_export.h>
 
 #include "../persondata.h"
 
@@ -37,7 +37,7 @@ class PersonDetailsViewPrivate;
  * Use PersonDetailsView to integrate a person's information
  * in the GUI.
  */
-class KPEOPLE_EXPORT PersonDetailsView : public QWidget
+class KPEOPLEWIDGETS_EXPORT PersonDetailsView : public QWidget
 {
     Q_OBJECT
 public:
